File: 0224_Basic Calculator/BasicCalculator.cs
public class Solution
    {
        public int Calculate(string s)
        {
            if (string.IsNullOrWhiteSpace(s)) return 0;

            int result = 0;
            int sign = 1;
            int num = 0;
            var stack = new Stack<int>();
            stack.Push(sign);

            foreach (var c in s)
            {
                if (char.IsDigit(c))
                {
                    num = num * 10 + (c - '0');
                }
                else if (c == '+' || c == '-')
                {
                    result += sign * num;
                    sign = stack.Peek() * (c == '+' ? 1 : -1);
                    num = 0;
                }
                else if (c == '(')
                {
                    stack.Push(sign);
                }
                else if (c == ')')
                {
                    stack.Pop();
                }
            }

            result += sign * num;
            return result;
        }
    }

File: 0224_Basic Calculator/BasicCalculator.java
class Solution {
    public int calculate(String s) {
        if(s.isEmpty()) return 0;
        
        int ans = 0;
        int sign = 1;
        int num = 0;
        Stack<Integer> stack = new Stack();
        stack.push(sign);
        
        for(char c : s.toCharArray()) {
            if(Character.isDigit(c)) {
                num *= 10;
                num += (c-'0');
            }else if(c == '(') {
                stack.push(sign);
            }else if(c == ')') {
                stack.pop();
            }else if(c == '+' || c == '-') {
                ans += sign * num;
                sign = stack.peek() * (c == '+' ? 1 : -1);
                num = 0;
            }
        }
        
        ans += sign * num;
        return ans;
    }
}